client/src/components/CreatePost.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useState } from "react";
+import CssBaseline from "@mui/material/CssBaseline";
+
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import Grid from "@mui/material/Grid";
+import Box from "@mui/material/Box";
+import Container from "@mui/material/Container";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Divider from "@mui/material/Divider";
+import Typography from "@mui/material/Typography";
+
+import Dialog from "@mui/material/Dialog";
+import DialogContent from "@mui/material/DialogContent";
+import DialogActions from "@mui/material/DialogActions";
+
+import PreviewPopUp from "./PreviewPopUp";
+import { DialogTitle } from "@mui/material";
+
+const theme = createTheme();
+
+function CreatePost() {
+  // create hooks for input fields
+  const [title, setTitle] = useState("");
+  const [body, setBody] = useState("");
+  const [file, setFile] = useState("");
+  const [tags, setTags] = useState("");
+  const [eventCapacity, setEventCapacity] = useState("");
+  const [eventLocation, setEventLocation] = useState("");
+
+  const [openPopup, setOpenPopup] = useState(false);
```

Comment:
If the actual popup is in `PreviewPopUp` then we should not be defining this here, the same idea applies to handleClose

Reply:
PreviewPopUp is just the component for the pop up, in order to trigger it, we require the hooks that need to be present in `CreatePost.tsx` because that's where the actual "preview" button is.

Reply:
Fair enough. Request a re-review from me once you go over the other comments